docs/book/compatiblity_matrix.md

```diff
@@ -1,16 +1,20 @@
 # Compatibility Matrix for vSphere CSI Driver
 
-- vSphere CSI driver is released with specific version of CSI sidecar containers which has the minimum kubernetes version requirement.
+- vSphere CSI driver is released with specific version of CSI sidecar containers which has the minimum and maximum kubernetes version requirement.
 
-| vSphere CSI Driver | Minimum Kubernetes Version |
-|--------------------|----------------------------|
-| v1.0.2             | v1.14.0                    |
-| v1.0.3             | v1.14.0                    |
-| v2.0.0             | v1.16.0                    |
-| v2.0.1             | v1.17.0                    |
-| v2.1.0             | v1.17.0                    |
+Note:
 
-- vSphere CSI driver is compatible with vSphere 67u3, vSphere 7.0 and 7.0u1.
+- vmware supports the deprecated release of the driver until End Of Life date.
+
+| vSphere CSI Driver | Minimum Kubernetes Version | Maximum Kubernetes Version            | Deprecated | End of Life  |
+|--------------------|----------------------------|---------------------------------------| -----------|--------------|
+| [v2.1.0](./releases/v2.1.0.md)             | v1.17.0                    | v1.19.0       | No         | -            |
+| [v2.0.1](./releases/v2.0.1.md)             | v1.17.0                    | v1.19.0       | Yes        | January 2022 |
+| [v2.0.0](./releases/v2.0.0.md)             | v1.16.0                    | v1.18.0       | Yes        | January 2022 |
+| [v1.0.3](./releases/v1.0.3.md)             | v1.14.0                    | v1.16.0       | Yes        | June 2021    |
+| [v1.0.2](./releases/v1.0.2.md)             | v1.14.0                    | v1.16.0       | Yes        | January 2021 |
+
+- vSphere CSI driver is compatible with vSphere 67u3, vSphere 7.0 and 7.0u1. Note: vCenter and ESXi should be on the same release. We do not support newer vCenter with older ESXi.
```

Comment:
Its common the the vCenter is upgraded first and then the ESX hosts are upgraded. In some large production deployments, the ESX upgrade takes time. Can you reword this to say that both the vCenter and ESX need to be at a version to use all the features? If ESX is old, then the CSI features would be limited to what the old ESX version can support.

Reply:
mentioned following

> If you have newer vCenter version but older ESXi hosts, new features we have added in the vCenter will not work if ESXi is not upgraded to newer version.